BDArmory/FX/ExplosionFX.cs

```diff
@@ -472,8 +468,13 @@ private bool IsInLineOfSight(Part part, Part explosivePart, out RaycastHit hit,
                         }
                         if (FlightGlobals.currentMainBody != null && hit.collider.gameObject == FlightGlobals.currentMainBody.gameObject) return false; // Terrain hit. Full absorption. Should avoid NREs in the following.
                         var partHP = partHit.Damage();
-						if (ProjectileUtils.IsArmorPart(partHit)) partHP = 100;
-                        var partArmour = partHit.GetArmorThickness();
+			if (ProjectileUtils.IsArmorPart(partHit)) partHP = 100;
+                        //var partArmour = partHit.GetArmorThickness();
+			var thickness = partHit.Modules.GetModule<HitpointTracker>();
+                        float partArmour = thickness.Armor;
+                        if (float.IsNaN(partArmour)) partArmour = 1;
+                        partArmour *= thickness.Density * thickness.Strength; //placeholder
+			
```

Comment:
density*strength placeholder for now; this is where a armor material modifier would be best suited to be added; as here you'd get the the total cumulative thickness counting discrete material mods, later in the code you'd have to settle for a homogeneous material thickness,